src/main/scala/ai/privado/policyEngine/PolicyExecutor.scala

```diff
@@ -207,6 +211,23 @@ class PolicyExecutor(
         }
       })
       .toSet
+    if (sinkFilters.sinkType.nonEmpty)
+      matchingSinkIds =
+        matchingSinkIds.filter(ruleCache.getRuleInfo(_).get.id.toLowerCase.contains(sinkFilters.sinkType.toLowerCase))
+
+    if (sinkFilters.domains.nonEmpty) {
+      matchingSinkIds = matchingSinkIds.filter { sinkId =>
+        val ruleInfo = ruleCache.getRuleInfo(sinkId)
+        println(ruleInfo.exists(info => info.domains.intersect(sinkFilters.domains).nonEmpty))
+        ruleInfo.exists(info => info.domains.intersect(sinkFilters.domains).nonEmpty)
+      }
+    }
+
+    if (sinkFilters.name.nonEmpty)
+      matchingSinkIds =
+        matchingSinkIds.filter(ruleCache.getRuleInfo(_).get.name.toLowerCase.contains(sinkFilters.name.toLowerCase))
```

Comment:
we can remove toLowerCase. And this should be a regex match not Contains

Reply:
DONE